Support of root list element

Hi, 

I'm testing `ryl` against existing yamllint valid files.

I've extracted the `.ryl.toml` file from the previous `.yamllint` file.

```sh
$ uvx ryl --migrate-configs --migrate-write 
$ cat .ryl.toml 
ignore = [
    "**/secret.yaml",
    "**/secret.yml",
    "**/assets/",
]
yaml-files = [
    "*.yaml",
    "*.yml",
    ".yamllint",
]

[rules]
anchors = "enable"
braces = "enable"
colons = "enable"
commas = "enable"
document-end = "disable"
empty-lines = "enable"
empty-values = "disable"
float-values = "disable"
hyphens = "enable"
indentation = "enable"
key-duplicates = "enable"
key-ordering = "disable"
new-line-at-end-of-file = "enable"
new-lines = "enable"
octal-values = "disable"
quoted-strings = "disable"
trailing-spaces = "enable"

[rules.brackets]
level = "warning"
max-spaces-inside = 1

[rules.comments]
level = "warning"

[rules.comments-indentation]
level = "warning"

[rules.document-start]
level = "warning"

[rules.line-length]
level = "warning"
max = 120

[rules.truthy]
level = "warning"

```

Some of my yaml files does not have a `root` (dict) element, but instead contains a list. ex: 

```yaml
---
# Foo bar
- name: "Foo"
- name: "Bar"

```
This kind of file raise issues with ryl : 
```sh
$ uvx ryl -c .ryl.toml ./test.yaml 
./test.yaml
  4:1       error    wrong indentation: expected 2 but found 0  (indentation)
```

The list element expect to be indented, but as root it cannot...

Thanks for your help !

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -68,11 +68,11 @@ repos:
         files: package.json
         pass_filenames: false
   - repo: https://github.com/rvben/rumdl-pre-commit
-    rev: v0.1.66
+    rev: v0.1.69
     hooks:
       - id: rumdl
   - repo: https://github.com/charliermarsh/ruff-pre-commit
-    rev: v0.15.9
+    rev: v0.15.10
     hooks:
       - id: ruff-check
         args: [--fix, --exit-non-zero-on-fix, --show-fixes]
@@ -80,7 +80,7 @@ repos:
       - id: ruff-format
         types_or: [python, pyi, jupyter]
   - repo: https://github.com/owenlamont/ryl-pre-commit
-    rev: v0.4.1
+    rev: v0.5.0
     hooks:
       - id: ryl
   - repo: https://github.com/adhtruong/mirrors-typos

--- a/.ryl.toml
+++ b/.ryl.toml
@@ -1,0 +1,47 @@
+yaml-files = [
+    "*.yaml",
+    "*.yml",
+    ".yamllint",
+]
+
+[rules]
+anchors = "enable"
+comments = "disable"
+comments-indentation = "disable"
+document-end = "disable"
+document-start = "disable"
+empty-values = "disable"
+float-values = "disable"
+key-duplicates = "enable"
+key-ordering = "disable"
+line-length = "disable"
+new-line-at-end-of-file = "enable"
+new-lines = "enable"
+octal-values = "disable"
+quoted-strings = "disable"
+trailing-spaces = "enable"
+truthy = "disable"
+
+[rules.braces]
+level = "warning"
+max-spaces-inside = 1
+
+[rules.brackets]
+level = "warning"
+max-spaces-inside = 1
+
+[rules.colons]
+level = "warning"
+
+[rules.commas]
+level = "warning"
+
+[rules.empty-lines]
+level = "warning"
+
+[rules.hyphens]
+level = "warning"
+
+[rules.indentation]
+indent-sequences = "consistent"
+level = "warning"

--- a/.yamllint
+++ b/.yamllint
@@ -1,4 +1,0 @@
-extends: relaxed
-
-rules:
-  line-length: disable

--- a/AGENTS.md
+++ b/AGENTS.md
@@ -161,7 +161,7 @@ sticking to the quick-status step above.
   character vs byte offset logic.
 - Use meaningful function and variable names in tests—comments are discouraged.
 - `#[cfg(test)]` modules inside `src/` is forbidden; add coverage through integration
-   tests in `tests/` so LLVM regions stay unique.
+  tests in `tests/` so LLVM regions stay unique.
 
 ## Release Checklist
 

--- a/README.md
+++ b/README.md
@@ -198,7 +198,7 @@ Example benchmark figure (5x5 matrix, 5 runs per point):
 - Presets and extends: supports yamllint’s built-in `default`, `relaxed`, and
   `empty` via `extends`. Rule maps are deep-merged; scalars/sequences overwrite.
 - TOML preset examples: see
-  [docs/config-presets.md](/home/owen/Code/ryl_repos/ryl/docs/config-presets.md)
+  [docs/config-presets.md](https://github.com/owenlamont/ryl/blob/main/docs/config-presets.md)
   for `default`/`relaxed` equivalents.
 
 Example TOML config (`.ryl.toml`):

--- a/src/rules/indentation.rs
+++ b/src/rules/indentation.rs
@@ -221,6 +221,7 @@ impl<'a> Analyzer<'a> {
         }
 
         if analysis.is_mapping_key()
+            && (!analysis.is_sequence_entry() || pushing_child)
             && let Some(ctx) = self.contexts.last_mut()
         {
             ctx.kind = analysis.context_kind();

--- a/src/rules/indentation.rs
+++ b/src/rules/indentation.rs
@@ -106,6 +106,7 @@ struct Analyzer<'a> {
     indent_seq: IndentSequencesRuntime,
     pending_child: Option<ContextKind>,
     multiline: Option<MultilineState>,
+    active_sequence_mapping_parent: Option<SequenceMappingParent>,
     compact_sequence_parent_indent: Option<usize>,
     compact_flow_mapping: Option<CompactFlowMapping>,
     prev_line_kind: Option<LineKind>,
@@ -127,6 +128,7 @@ impl<'a> Analyzer<'a> {
             indent_seq: IndentSequencesRuntime::new(cfg.indent_sequences),
             pending_child: None,
             multiline: None,
+            active_sequence_mapping_parent: None,
             compact_sequence_parent_indent: None,
             compact_flow_mapping: None,
             prev_line_kind: None,
@@ -145,25 +147,7 @@ impl<'a> Analyzer<'a> {
     fn process_line(&mut self, line_number: usize, raw: &str) {
         let line = raw.trim_end_matches(['\r', '\n']);
         let (indent, content) = split_indent(line);
-        if self
-            .compact_sequence_parent_indent
-            .is_some_and(|parent| indent <= parent)
-        {
-            self.compact_sequence_parent_indent = None;
-        }
-        if self
-            .compact_flow_mapping
-            .is_some_and(|state| indent <= state.parent_indent)
-        {
-            self.compact_flow_mapping = None;
-        }
-
-        if let Some(state) = &self.multiline
-            && indent <= state.base_indent
-            && !content.trim().is_empty()
-        {
-            self.multiline = None;
-        }
+        self.reset_transient_state(indent, content);
 
         if content.trim().is_empty() {
             self.prev_line_kind = Some(LineKind::Other);
@@ -226,7 +210,43 @@ impl<'a> Analyzer<'a> {
         {
             ctx.kind = analysis.context_kind();
         }
+        self.update_post_analysis_state(indent, content, analysis);
+    }
 
+    fn reset_transient_state(&mut self, indent: usize, content: &str) {
+        if self
+            .compact_sequence_parent_indent
+            .is_some_and(|parent| indent <= parent)
+        {
+            self.compact_sequence_parent_indent = None;
+        }
+        if self
+            .compact_flow_mapping
+            .is_some_and(|state| indent <= state.parent_indent)
+        {
+            self.compact_flow_mapping = None;
+        }
+
+        if let Some(state) = &self.multiline
+            && indent <= state.base_indent
+            && !content.trim().is_empty()
+        {
+            self.multiline = None;
+        }
+
+        if self.active_sequence_mapping_parent.is_some_and(|state| {
+            !content.trim().is_empty() && indent <= state.owner_indent
+        }) {
+            self.active_sequence_mapping_parent = None;
+        }
+    }
+
+    fn update_post_analysis_state(
+        &mut self,
+        indent: usize,
+        content: &str,
+        analysis: LineAnalysis,
+    ) {
         if analysis.starts_multiline {
             self.multiline = Some(MultilineState::new(indent));
         }
@@ -235,6 +255,13 @@ impl<'a> Analyzer<'a> {
             self.pending_child = Some(analysis.context_kind());
         } else {
             self.pending_child = None;
+        }
+
+        if analysis.is_sequence_entry() && analysis.opens_child_context() {
+            self.active_sequence_mapping_parent = Some(SequenceMappingParent {
+                owner_indent: indent,
+                parent_indent: indent.saturating_add(analysis.sequence_offset),
+            });
         }
 
         if is_compact_sequence_start(content) {
@@ -350,8 +377,15 @@ impl<'a> Analyzer<'a> {
     }
 
     fn check_sequence_indent(&mut self, indent: usize, line_number: usize) {
-        let Some((ctx_index, parent_indent)) = self.find_mapping_parent_indent(indent)
-        else {
+        let (ctx_index, parent_indent) = if let Some((ctx_index, parent_indent)) =
+            self.find_mapping_parent_indent(indent)
+        {
+            (Some(ctx_index), parent_indent)
+        } else if let Some(state) = self.active_sequence_mapping_parent
+            && indent > state.owner_indent
+        {
+            (None, state.parent_indent)
+        } else {
             return;
         };
 
@@ -361,17 +395,30 @@ impl<'a> Analyzer<'a> {
             .expected_step()
             .map(|step| parent_indent.saturating_add(step));
 
-        let state = &mut self.sequence_expectations[ctx_index];
-        if let Some(message) =
-            self.indent_seq
-                .check(parent_indent, indent, is_indented, expected, state)
-        {
-            self.diagnostics.push(Violation {
-                line: line_number,
-                column: indent + 1,
-                message,
-            });
-        }
+        let Some(message) = (match ctx_index {
+            Some(ctx_index) => self.indent_seq.check(
+                parent_indent,
+                indent,
+                is_indented,
+                expected,
+                &mut self.sequence_expectations[ctx_index],
+            ),
+            None => self.indent_seq.check(
+                parent_indent,
+                indent,
+                is_indented,
+                expected,
+                &mut None,
+            ),
+        }) else {
+            return;
+        };
+
+        self.diagnostics.push(Violation {
+            line: line_number,
+            column: indent + 1,
+            message,
+        });
     }
 }
 
@@ -388,6 +435,12 @@ struct CompactFlowMapping {
 }
 
 #[derive(Debug, Clone, Copy)]
+struct SequenceMappingParent {
+    owner_indent: usize,
+    parent_indent: usize,
+}
+
+#[derive(Debug, Clone, Copy)]
 enum ContextKind {
     Root,
     Mapping { sequence_offset: usize },
@@ -400,6 +453,7 @@ struct LineAnalysis {
     kind: LineKind,
     starts_multiline: bool,
     is_sequence_entry: bool,
+    sequence_offset: usize,
 }
 
 #[derive(Debug, Clone, Copy)]
@@ -419,10 +473,15 @@ impl LineAnalysis {
         let trimmed = core.trim();
         let is_sequence_entry = is_sequence_entry(trimmed);
         let (is_mapping_key, opens_block) = classify_mapping(trimmed);
+        let sequence_offset = if is_mapping_key {
+            sequence_prefix_width(trimmed)
+        } else {
+            0
+        };
         let kind = if is_mapping_key {
             LineKind::Mapping {
                 opens_block,
-                sequence_offset: sequence_prefix_width(trimmed),
+                sequence_offset,
             }
         } else if is_sequence_entry {
             LineKind::Sequence
@@ -434,6 +493,7 @@ impl LineAnalysis {
             kind,
             starts_multiline,
             is_sequence_entry,
+            sequence_offset,
         }
     }
 

--- a/tests/rule_indentation.rs
+++ b/tests/rule_indentation.rs
@@ -241,6 +241,28 @@ fn skips_blank_lines_and_top_level_sequence_entries() {
 }
 
 #[test]
+fn top_level_sequence_of_inline_mappings_is_allowed() {
+    let cfg = config(SpacesSetting::Fixed(2), IndentSequencesSetting::True, false);
+    let yaml = "- name: Foo\n- name: Bar\n";
+    assert!(indentation::check(yaml, &cfg).is_empty());
+}
+
+#[test]
+fn indented_top_level_sequence_of_inline_mappings_hits_fallback_parent_indent() {
+    let cfg = config(SpacesSetting::Fixed(2), IndentSequencesSetting::True, false);
+    let yaml = "  - key: Foo\n  - key: Bar\n";
+    let hits = indentation::check(yaml, &cfg);
+    assert_eq!(
+        hits,
+        vec![Violation {
+            line: 2,
+            column: 3,
+            message: "wrong indentation: expected 4 but found 2".to_string(),
+        }]
+    );
+}
+
+#[test]
 fn reports_misaligned_mapping_with_consistent_spacing() {
     let cfg = config(
         SpacesSetting::Consistent,
@@ -352,12 +374,10 @@ fn top_level_indented_plain_scalar_is_permitted() {
 }
 
 #[test]
-fn sequence_of_mappings_reports_incorrect_dedent() {
+fn sequence_of_mappings_can_dedent_to_root() {
     let cfg = config(SpacesSetting::Fixed(2), IndentSequencesSetting::True, false);
     let yaml = "- key:\n    - nested\n- other\n";
-    let hits = indentation::check(yaml, &cfg);
-    assert_eq!(hits.len(), 1, "unexpected diagnostics: {hits:?}");
-    assert!(hits[0].message.contains("expected 2 but found 0"));
+    assert!(indentation::check(yaml, &cfg).is_empty());
 }
 
 #[test]

--- a/tests/rule_indentation.rs
+++ b/tests/rule_indentation.rs
@@ -248,18 +248,10 @@ fn top_level_sequence_of_inline_mappings_is_allowed() {
 }
 
 #[test]
-fn indented_top_level_sequence_of_inline_mappings_hits_fallback_parent_indent() {
+fn nested_inline_mapping_sequence_entries_share_parent_indent() {
     let cfg = config(SpacesSetting::Fixed(2), IndentSequencesSetting::True, false);
-    let yaml = "  - key: Foo\n  - key: Bar\n";
-    let hits = indentation::check(yaml, &cfg);
-    assert_eq!(
-        hits,
-        vec![Violation {
-            line: 2,
-            column: 3,
-            message: "wrong indentation: expected 4 but found 2".to_string(),
-        }]
-    );
+    let yaml = "root:\n  - key: Foo\n  - key: Bar\n";
+    assert!(indentation::check(yaml, &cfg).is_empty());
 }
 
 #[test]
@@ -378,6 +370,36 @@ fn sequence_of_mappings_can_dedent_to_root() {
     let cfg = config(SpacesSetting::Fixed(2), IndentSequencesSetting::True, false);
     let yaml = "- key:\n    - nested\n- other\n";
     assert!(indentation::check(yaml, &cfg).is_empty());
+}
+
+#[test]
+fn sequence_entry_mapping_requires_nested_sequence_indent_when_enabled() {
+    let cfg = config(SpacesSetting::Fixed(2), IndentSequencesSetting::True, false);
+    let yaml = "- key:\n  - nested\n";
+    let hits = indentation::check(yaml, &cfg);
+    assert_eq!(
+        hits,
+        vec![Violation {
+            line: 2,
+            column: 3,
+            message: "wrong indentation: expected 4 but found 2".to_string(),
+        }]
+    );
+}
+
+#[test]
+fn top_level_indented_inline_mapping_sequence_uses_fallback_parent_indent() {
+    let cfg = config(SpacesSetting::Fixed(2), IndentSequencesSetting::True, false);
+    let yaml = "  - key: Foo\n  - key: Bar\n";
+    let hits = indentation::check(yaml, &cfg);
+    assert_eq!(
+        hits,
+        vec![Violation {
+            line: 2,
+            column: 3,
+            message: "wrong indentation: expected 4 but found 2".to_string(),
+        }]
+    );
 }
 
 #[test]

--- a/tests/yamllint_compat_indentation.rs
+++ b/tests/yamllint_compat_indentation.rs
@@ -313,6 +313,46 @@ fn root_sequence_of_inline_mappings_matches_yamllint() {
 }
 
 #[test]
+fn sequence_entry_mapping_nested_sequence_matches_yamllint() {
+    ensure_yamllint_installed();
+
+    let dir = tempdir().unwrap();
+
+    let cfg_path = dir.path().join("cfg.yaml");
+    fs::write(
+        &cfg_path,
+        "rules:\n  document-start: disable\n  indentation:\n    spaces: 2\n    indent-sequences: true\n",
+    )
+    .unwrap();
+
+    let yaml_path = dir.path().join("input.yaml");
+    fs::write(&yaml_path, "- key:\n  - nested\n").unwrap();
+
+    let exe = env!("CARGO_BIN_EXE_ryl");
+
+    for scenario in SCENARIOS {
+        let mut ryl_cmd = build_ryl_command(exe, scenario.ryl_format);
+        ryl_cmd.arg("-c").arg(&cfg_path).arg(&yaml_path);
+        let (ryl_code, ryl_msg) = capture_with_env(ryl_cmd, scenario.envs);
+
+        let mut yam_cmd = build_yamllint_command(scenario.yam_format);
+        yam_cmd.arg("-c").arg(&cfg_path).arg(&yaml_path);
+        let (yam_code, yam_msg) = capture_with_env(yam_cmd, scenario.envs);
+
+        assert_eq!(
+            ryl_code, yam_code,
+            "exit mismatch for nested sequence-entry mapping ({})",
+            scenario.label
+        );
+        assert_eq!(
+            ryl_msg, yam_msg,
+            "diagnostics mismatch for nested sequence-entry mapping ({})",
+            scenario.label
+        );
+    }
+}
+
+#[test]
 fn readthedocs_style_indentation_matches_yamllint() {
     ensure_yamllint_installed();
 

--- a/tests/yamllint_compat_indentation.rs
+++ b/tests/yamllint_compat_indentation.rs
@@ -272,6 +272,47 @@ fn compact_nested_sequence_indentation_matches_yamllint() {
 }
 
 #[test]
+fn root_sequence_of_inline_mappings_matches_yamllint() {
+    ensure_yamllint_installed();
+
+    let dir = tempdir().unwrap();
+
+    let cfg_path = dir.path().join("cfg.yaml");
+    fs::write(
+        &cfg_path,
+        "rules:\n  document-start: disable\n  indentation:\n    spaces: 2\n",
+    )
+    .unwrap();
+
+    let yaml_path = dir.path().join("input.yaml");
+    fs::write(&yaml_path, "---\n# Foo bar\n- name: Foo\n- name: Bar\n").unwrap();
+
+    let exe = env!("CARGO_BIN_EXE_ryl");
+
+    for scenario in SCENARIOS {
+        let mut ryl_cmd = build_ryl_command(exe, scenario.ryl_format);
+        ryl_cmd.arg("-c").arg(&cfg_path).arg(&yaml_path);
+        let (ryl_code, ryl_msg) = capture_with_env(ryl_cmd, scenario.envs);
+
+        let mut yam_cmd = build_yamllint_command(scenario.yam_format);
+        yam_cmd.arg("-c").arg(&cfg_path).arg(&yaml_path);
+        let (yam_code, yam_msg) = capture_with_env(yam_cmd, scenario.envs);
+
+        assert_eq!(
+            yam_code, 0,
+            "yamllint should accept input ({})",
+            scenario.label
+        );
+        assert_eq!(
+            ryl_code, 0,
+            "ryl should accept input ({})\n{}",
+            scenario.label, ryl_msg
+        );
+        assert_eq!(ryl_msg, yam_msg, "output mismatch ({})", scenario.label);
+    }
+}
+
+#[test]
 fn readthedocs_style_indentation_matches_yamllint() {
     ensure_yamllint_installed();
 

